src/form/iv.jl

```diff
@@ -566,3 +378,48 @@ function _objective_min_fuel_and_flow_cost_polynomial_linquad(pm::AbstractIVRMod
         for (n, nw_ref) in nws(pm))
     )
 end
+
+
+"adds pg_cost variables and constraints"
+function objective_variable_pg_cost(pm::AbstractIVRModel)
+    for (n, nw_ref) in nws(pm)
+        gen_lines = calc_cost_pwl_lines(nw_ref[:gen])
+
+        #to avoid function calls inside of @NLconstraint
+        pg_cost = var(pm, n)[:pg_cost] = JuMP.@variable(pm.model,
+            [i in ids(pm, n, :gen)], base_name="$(n)_pg_cost",
+        )
+        nc = length(conductor_ids(pm, n))
+
+        # gen pwl cost
+        for (i, gen) in nw_ref[:gen]
+            pg = [var(pm, n, c, :pg, i) for c in conductor_ids(pm, n)]
+            for line in gen_lines[i]
+                JuMP.@NLconstraint(pm.model, pg_cost[i] >= line.slope*sum(pg[c] for c in 1:nc) + line.intercept)
```

Comment:
Is `@NLconstraint` this the only difference here?  If so, why isn't `sum(pg[c] for c in 1:nc)` a quadratic expression?

Reply:
:pg contains @NLexpressions, which can be nested into @NLconstraints only.